### PR TITLE
Distinguish date and datetime data type in spss serialization

### PR DESCRIPTION
**Pull request by ewelinagr**

### CODE REVIEW

File: transmart-rest-api/src/main/groovy/org/transmartproject/rest/serialization/tabular/TabularResultSPSSSerializer.groovy [hunk 1]
```diff
@@ -121,25 +122,54 @@ class TabularResultSPSSSerializer implements TabularResultSerializer {
...
+            return value.toString()
+        }
+    }
+
+    private String formatDateRowValue(Date value, VariableMetadata metadata) {
+        if (metadata.type == VariableDataType.DATETIME) {
+            synchronized (DATETIME_FORMAT) {
+                return DATETIME_FORMAT.format(value)
+            }
+        } else {
```
Comment (forus) [inline transmart-rest-api/src/main/groovy/org/transmartproject/rest/serialization/tabular/TabularResultSPSSSerializer.groovy]: maybe it makes more sense to return datetime format by default?

File: transmart-rest-api/src/test/groovy/org/transmartproject/rest/serialization/tabular/TabularResultSPSSSerializerSpec.groovy [hunk 2]
```diff
@@ -266,6 +273,66 @@ class TabularResultSPSSSerializerSpec extends Specification {
...
+        def zipIn = new ZipInputStream(bytes)
+        def entry = zipIn.nextEntry
+        entry.name == 'data.tsv'
+        def tsvReader = new CSVReaderBuilder(new InputStreamReader(zipIn))
+                .withCSVParser(new CSVParserBuilder().withSeparator(AbstractTSVSerializer.COLUMN_SEPARATOR).build())
+                .build()
+        def lines = tsvReader.readAll()
+
+        then:
+        lines.isEmpty()
```
Comment (forus) [inline transmart-rest-api/src/test/groovy/org/transmartproject/rest/serialization/tabular/TabularResultSPSSSerializerSpec.groovy]: I guess it's work in progress

File: transmart-rest-api/src/main/groovy/org/transmartproject/rest/serialization/tabular/TabularResultSPSSSerializer.groovy (hunk 1 above)
Comment (ewelinagr) [reply transmart-rest-api/src/main/groovy/org/transmartproject/rest/serialization/tabular/TabularResultSPSSSerializer.groovy]: good point, I changed it

File: transmart-rest-api/src/test/groovy/org/transmartproject/rest/serialization/tabular/TabularResultSPSSSerializerSpec.groovy (hunk 2 above)
Comment (ewelinagr) [reply transmart-rest-api/src/test/groovy/org/transmartproject/rest/serialization/tabular/TabularResultSPSSSerializerSpec.groovy]: yes, but now it is finished

**Comment by forus** [comment]
Nice job!

**Pull request merged**